### PR TITLE
script to parallelize uploads (GDEV-1502)

### PR DESCRIPTION
**Pull request by KerstenBreuer**
```
Added a script that takes a tsv file that lists files to be uploaded as input
and triggers the upload of multiple files in parallel.
```

### CODE REVIEW

**Review by PMenges — APPROVED** [review]
- Tested and worked on the odcf cluster.

**Review by PMenges — DISMISSED** [review]

File: src/otp_tsv_to_upload.py [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+        if not file.path.exists():
+            raise RuntimeError(
+                f"The file at the following path does not exist: {file.path}"
+            )
+
+    return files
+
+
+def check_file_upload(file: FileMetadata, output_dir: Path) -> bool:
+    """Returns true if the file was already uploaded. Returns falls otherwise."""
```
Comment (MoritzHahn1337) [inline src/otp_tsv_to_upload.py:70]:
```suggestion
    """Returns true if the file was already uploaded. Returns false otherwise."""
```

**Review by MoritzHahn1337 — APPROVED** [review]
Looks good

**Review by mephenor — APPROVED** [review]

File: src/otp_tsv_to_upload.py [hunk 2]
```diff
@@ -0,0 +1,219 @@
...
+            FileMetadata(
+                path=Path(line.split("\t")[0].strip()),
+                alias=line.split("\t")[1].strip(),
+            )
+            for line in tsv_file.readlines()
+            if line != ""
+        ]
+
+    for file in files:
+        if not file.path.exists():
```
[the remark this answers is not in the document]
Comment (KerstenBreuer) [reply src/otp_tsv_to_upload.py:61]: Good point. Let's make the absolute.

File: src/otp_tsv_to_upload.py [hunk 3]
```diff
@@ -50,7 +50,7 @@ def load_file_metadata(otp_tsv: Path) -> list[FileMetadata]:
     with open(otp_tsv, "r", encoding="utf-8") as tsv_file:
         files = [
             FileMetadata(
-                path=Path(line.split("\t")[0].strip()),
+                path=Path(line.split("\t")[0].strip()).absolute(),
```
Comment (mephenor) [inline src/otp_tsv_to_upload.py:53]:
```suggestion
                path=Path(line.split("\t")[0].strip()).resolve(),
```
Apparently absolute is now fine from Python 3.11 on, but undocumented + untested before... so your choice

**Review by mephenor — DISMISSED** [review]

**Review by PMenges — APPROVED** [review]
Tested and worked on the odcf cluster.

**Review by mephenor — APPROVED** [review]

**Pull request merged**